src/NAR/RunTime

```diff
@@ -0,0 +1,77 @@
+package NAR;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class RunTime {
+
+	public ArrayList<String> listOfLines(String str) {
+
+		ArrayList<String> lines = new ArrayList<>();
+		Scanner scanner = new Scanner(str);
+
+		while (scanner.hasNextLine()) {
+
+			lines.add(scanner.nextLine());
+		}
+
+		scanner.close();
+		return lines;
+	}
+
+	public String splitLine(String str) {
+
+		String line = str.replaceAll("\\s +", " ");
+
+		if (isLengthGreaterThanZero(line)) {
+
+			if (isFirstCharWhiteSpace(line)) {
+
+				line = line.replaceFirst("\\ +", "");
+			}
+		}
+
+		return line;
+	}
+
+	private static boolean isFirstCharWhiteSpace(String str) {
+
+		char fChar = str.charAt(0);
+		return Character.isWhitespace(fChar);
+	}
+
+	public static boolean isLengthGreaterThanZero(String str) {
+
+		return str.length() > 0 && str != null;
```

Comment:
you should be checking for null first and then for the length because right now the code might throw a null pointer exception